GrammarNazi.App/HostedServices/TwitterBotHostedService.cs

```diff
@@ -112,21 +112,41 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
                         var correctionString = messageBuilder.ToString();
 
-                        // TODO: Separate corrections in various tweets when max characters exceeded https://github.com/nminaya/grammar-nazi-bot/issues/73
-                        if (correctionString.Length - mentionedUsers.Length > Defaults.TwitterTextMaxLength)
-                            continue;
-
                         _logger.LogInformation($"Sending reply to: {tweet.CreatedBy.ScreenName}");
-                        var publishTweetParameters = new PublishTweetParameters(correctionString)
-                        {
-                            InReplyToTweetId = tweet.Id
-                        };
-                        var replyTweet = await _twitterClient.Tweets.PublishTweetAsync(publishTweetParameters);
 
-                        if (replyTweet != null)
+                        if (correctionString.Length < Defaults.TwitterTextMaxLength)
+                        {
+                            var publishTweetParameters = new PublishTweetParameters(correctionString)
+                            {
+                                InReplyToTweetId = tweet.Id
+                            };
+                            var replyTweet = await _twitterClient.Tweets.PublishTweetAsync(publishTweetParameters);
+
+                            if (replyTweet != null)
+                            {
+                                _logger.LogInformation("Reply sent successfuly");
+                                await _twitterLogService.LogReply(tweet.Id, replyTweet.Id);
+                            }
+                        }
+                        else
                         {
-                            _logger.LogInformation("Reply sent successfuly");
-                            await _twitterLogService.LogReply(tweet.Id, replyTweet.Id);
+                            var replyTweets = correctionString.SplitInParts(Defaults.TwitterTextMaxLength);
+
+                            foreach (var reply in replyTweets)
+                            {
+                                var correctionStringSplitted = $"@{tweet.CreatedBy.ScreenName} {mentionedUsers} {reply}";
```

Comment:
@Jadhielv In the first cicle of this foreach, the ```reply``` variable already will have the mentioned users, since this is done in line 105.

The mentions should be added from the second cycle and up.

Reply:
@Jadhielv You can do something like:

```cs
...
foreach (var (reply, index) in replyTweets.Select((reply, index) => (reply, index)))
{
    var correctionStringSplitted = index == 0 ? reply : $"@{tweet.CreatedBy.ScreenName} {mentionedUsers} {reply}";
    var publishTweetsParameters = new PublishTweetParameters(correctionStringSplitted)
    {
...
```

Reply:
Done @nminaya